cgen error for `struct_field shared map[string]&Abc`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.0 286d397.65a493d
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2023-08-10 05:49:14

vroot: OK, value: /v/vnew
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.41.0
Git vroot status: weekly.2023.32-17-ge3b0dfbf
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg shared_map_default_init_cgen_error.v`
```v
struct Abc {
	f shared map[string]&Abc
}

fn main() {
	_ := Abc{}
}

```

**What did you expect to see?**

a compiled program

**What did you see instead?**
```
==================
/tmp/v_1000/shared_map_default_init_cgen_error.16775563098376142011.tmp.c:18008: error: '__shared__Map_string_main__Abc' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

The error with clang is more informative:
```
/tmp/v_1000/shared_map_default_init_cgen_error.6868054527787955160.tmp.c:18025:67: error: expected expression
        {main__Abc _ = ((main__Abc){.f = (__shared__Map_string_main__Abc*)__dup_shared_map(&(__shared__Map_string_main__Abc){.mtx = {0}, .val =new_map(sizeof(string), sizeof(main__Abc*), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string)}, sizeof(__shared__Map_string_main__Abc)),});}
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6338,7 +6338,8 @@ fn (mut g Gen) type_default(typ_ ast.Type) string {
 			}
 			init_str := 'new_map${noscan}(sizeof(${g.typ(info.key_type)}), sizeof(${g.typ(info.value_type)}), ${hash_fn}, ${key_eq_fn}, ${clone_fn}, ${free_fn})'
 			if typ.has_flag(.shared_f) {
-				mtyp := '__shared__Map_${key_typ.cname}_${g.table.sym(info.value_type).cname}'
+				mtyp := '__shared__Map_${key_typ.cname}_${g.typ(info.value_type).replace('*',
+					'_ptr')}'
 				return '(${mtyp}*)__dup_shared_map(&(${mtyp}){.mtx = {0}, .val =${init_str}}, sizeof(${mtyp}))'
 			} else {
 				return init_str

